arch/arc/core/prep_c.c

```diff
@@ -88,8 +88,13 @@ static void invalidate_dcache(void)
 static void adjust_vector_table_base(void)
 {
 #ifdef CONFIG_ARC_HAS_SECURE
+#ifdef CONFIG_ARC_SECURE_FIRMWARE
+/* only when processor in secure mode, the vector base should be
```

Comment:
Why? Normal mode has it's own vector table, shouldn't that be used (and updated) in Normal mode?